packages/core/src/fs/project-file.ts

```diff
@@ -77,13 +83,26 @@ export class ProjectFile implements FileDescription {
     await this.fs.mkdir(path.dirname(backupFileName), { recursive: true });
     if (this.#originalContent === undefined) {
       await this.fs.copyFile(this.name, backupFileName);
+      this.#mode = await this.getChmod(this.name);
     } else {
       await this.fs.writeFile(backupFileName, this.#originalContent);
     }
+    await this.setChmod(backupFileName);
     return backupFileName;
   }
 
   public get hasChanges(): boolean {
     return this.#currentContent !== this.#originalContent;
   }
+
+  private async getChmod(file: string): Promise<number | undefined> {
```

Comment:
This class uses ES privates, not typescript `private`. So instead of `private getChmod` we should use `#getChmod`. ES Privates are a recent addition to JavaScript, so they're natively supported.